src/fdcache.cpp

```diff
@@ -494,6 +502,117 @@ int PageList::GetUnloadedPages(fdpage_list_t& unloaded_list, off_t start, size_t
   return unloaded_list.size();
 }
 
+// [NOTE]
+// This method is called in advance when mixing POST and COPY in multi-part upload.
+// The minimum size of each part must be 5 MB, and the data area below this must be
+// downloaded from S3.
+// This method checks the current PageList status and returns the area that needs
+// to be downloaded so that each part is at least 5 MB.
+//
+bool PageList::GetLoadPageListForMultipartUpload(fdpage_list_t& dlpages)
+{
+  // compress before this processing
+  if(!Compress()){
+    return false;
+  }
+
+  size_t    before_loaded_size = 0;
+  bool      before_indigent    = false;
```

Comment:
What does indigent mean?

Reply:
Sorry, it means "not enough of 5MB", I will change the flag name to "before_insufficient".